testplan/common/utils/remote.py

```diff
@@ -135,3 +137,28 @@ def rm_cmd(path):
 def filepath_exist_cmd(path):
     """Checks if filepath exists."""
     return " ".join(["test", "-e", path])
+
+
+def calculate_pool_size_by_tasks(
+    tasks: Iterable[Task], plan_runtime_target: int
+):
+    _tasks = sorted(tasks, key=lambda task: task.weight, reverse=True)
+    max_weight = (
+        max(_tasks[0].weight, plan_runtime_target)
+        if _tasks
+        else plan_runtime_target
+    )
+    if max_weight > plan_runtime_target:
```

Comment:
It seems a bit redundant that we are doing both max and larger_than comparison. I think just the compare and update plan_runtime_target will look neater.